### PR TITLE
Added API documentation for the Release endpoint

### LINKED ISSUE #3985: Add API documentation for the Release endpoint
In https://warehouse.readthedocs.io/api-reference/json/ we have nearly no documentation of the Release endpoint https://warehouse.readthedocs.io/api-reference/json/#release . We should have a sample request header and response body as we do for the Project endpoint (also on that page). You can fix this in `docs/api-reference/json.rst` and basically copy the format from the Project endpoint sample in that same file.

I suggest you use the `requests` Python module to send a request to the endpoint for some specific PyPI project and release, and figure out what the request and response are, and then add the relevant bits of those to the document.

Once you have started changing that file, you will need to `make docs` to preview your changes locally.

---

**Good First Issue**: This issue is good for first time contributors. If you've already contributed to Warehouse, please work on [another issue without this label](https://github.com/pypa/warehouse/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+-label%3A%22good+first+issue%22) instead. If there is not a corresponding pull request for this issue, it is up for grabs. For directions for getting set up, see our [Getting Started Guide](https://warehouse.pypa.io/development/getting-started/). If you are working on this issue and have questions, please feel free to ask them here, [`#pypa-dev` on Freenode](https://webchat.freenode.net/?channels=%23pypa-dev), or the [pypa-dev mailing list](https://groups.google.com/forum/#!forum/pypa-dev).

### PR DESCRIPTION
**Pull request by JKiely**
Fixes #3985.

* Added an example response from the Release endpoint

* Changed example package used in Project endpoint

### CODE REVIEW

**Comment by di** [comment]
Going to cancel CI for this as we're hitting the limits for our org, but I will re-start it after I (or someone else) has gotten a chance to review this.

**Comment by brainwane** [comment]
@yeraydiazdiaz or @alanbato could you review this? Thanks!

**Comment by alanbato** [comment]
Looks good to me :+1:

File: docs/api-reference/json.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ Project
 
     .. code:: http
 
-        GET /pypi/pip/json HTTP/1.1
+        GET /pypi/visidata/json HTTP/1.1
```
Comment (di) [inline docs/api-reference/json.rst]: Any particular reason why we're changing the example project here?

Comment (brainwane) [reply docs/api-reference/json.rst]: I suggested this. I think that using `pip` as the example project is apt to cause at least a little confusion when less experienced developers run into problems and try to reason through what has gone wrong.

Comment (di) [reply docs/api-reference/json.rst]: Can we use [`sampleproject`](https://pypi.org/project/sampleproject/) instead then?

Comment (brainwane) [reply docs/api-reference/json.rst]: Great idea!

**Pull request merged**